### PR TITLE
Better mocking

### PR DESCRIPTION
**Pull request by AbdealiJK**

### CODE REVIEW

**Comment by AbdealiJK** [comment]
Codecov seems dead.

File: coalib/bears/Bear.py [hunk 1]
```diff
@@ -32,10 +32,14 @@ class Bear(Printer, LogPrinter):
...
     To indicate which languages your bear supports, just give it the
-    ``LANGUAGES`` value:
+    ``LANGUAGES`` value which can either be a string (if the bear supports
+    only 1 language) or a tuple of strings:
 
     >>> class SomeBear(Bear):
     ...     LANGUAGES = ('C', 'CPP','C#', 'D')
+
+    >>> class SomeBear(Bear):
+    ...     LANGUAGES = "MardDown"
```
Comment (sils1297) [inline coalib/bears/Bear.py]: I changed this to Markdown with lower case d in that bear, can we have the same here?

File: coalib/output/ConsoleInteraction.py [hunk 2]
```diff
@@ -637,12 +637,11 @@ def show_bear(console_printer, bear, sections, metadata):
...
 
-    show_enumeration(console_printer,
-                     "Supported languages:",
-                     bear.LANGUAGES,
-                     "  ",
-                     "The bear does not provide information about which "
-                     "languages it can analyze.")
+    show_enumeration(
+        console_printer, "Supported languages:",
+        [bear.LANGUAGES] if isinstance(bear.LANGUAGES, str) else bear.LANGUAGES,
```
Comment (sils1297) [inline coalib/output/ConsoleInteraction.py]: use tuple instead list?

Comment (AbdealiJK) [inline coalib/output/ConsoleInteraction.py]: I don't like tuples with 1 element as they're confusing - list is alright in this case right ?

Comment (sils1297) [inline coalib/output/ConsoleInteraction.py]: IMO we should really stop using lists so much, they're dangerous and mutable. tuples are light and safe.

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2076/commits/0b310e0c80d3b7142b739ee0129f9e919c56b167 https://github.com/coala-analyzer/coala/pull/2076/commits/8b5f6916ee70e83dd48a0e9ec0559d00827b928f https://github.com/coala-analyzer/coala/pull/2076/commits/6a6be29c406d562bb78eaed9c405e0f98c9df264

**Comment by sils1297** [comment]
37604e6 needs work

**Comment by AbdealiJK** [comment]
reack
e68d184 coalaDeleteOrigTest: Use unittest.mock
b50c7fb TestUtilities: Use unittest.mock
e8737fe coalaTest: Use bear_test_module() in coala -A test

**Comment by AbdealiJK** [comment]
reack

**Comment by AbdealiJK** [comment]
reack
5c6e32b coalaDeleteOrigTest: Use unittest.mock
778fa72 TestUtilities: Use unittest.mock
15de1bc coalaTest: Use bear_test_module() in coala -A test

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2076/commits/f51d880d37a5e11599fc880bd4bb2bd13a11d39a

**Comment by sils1297** [comment]
coverage is full

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**